cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -2192,49 +1809,16 @@ Slice::JavaGenerator::getTypeMetadata(const MetadataList& metadata, string& inst
 }
 
 bool
-Slice::JavaGenerator::hasTypeMetadata(const TypePtr& type, const MetadataList& localMetadata)
```

Comment:
Two simplifications:
1) This is only called for sequences, so we can eliminate alot of casting/checking of types.
2) This function included validation of the metadata, which was already being performed by the validator.